CONTRIBUTING.md

```diff
@@ -1,16 +1,256 @@
-# CONTRIBUTE
-This is a short version of the [Development Procedures](https://github.com/RIOT-OS/RIOT/wiki/Development-procedures).
+# Contributing to RIOT
 
-1. Check if your code follows the [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions). If the code does not comply these style rules, your code will not be merged.
+Thank you for your interest in contributing to RIOT! There are many ways to
+contribute, and we appreciate all of them. You can jump to the major sections
+of this document using the following links:
 
-2. The master branch should always be in a working state. The RIOT maintainers will create release tags based on this branch, whenever a milestone is completed.
+* [Feature Requests](#feature-requests)
+* [Bug Reports](#bug-reports)
+* [Pull Requests](#pull-requests)
+* [Writing Documentation](#writing-documentation)
 
-3. Comments on a pull request should be added to the request itself, and *not* to the commit.
+If you have questions, please send an email to users@riot-os.org or
+devel@riot-os.org mailing list or chat on [#riot-os][riot-chat].
 
-4. Keep commits to the point, e.g., don't add whitespace/typo fixes to other code changes. If changes are layered, layer the patches.
+As a reminder, all contributors are expected to follow our
+[Code of Conduct](CODE_OF_CONDUCT.md).
 
-5. Describe the technical detail of the change(s) as specific as possible.
+[riot-chat]: http://webchat.freenode.net?channels=riot-os
 
-6. Use [Labels](https://github.com/RIOT-OS/RIOT/wiki/Labels) to help classify pull requests and issues.
+## Feature Requests
+[feature-requests]: #feature-requests
 
-7. Please adhere to our [code of conduct](https://github.com/RIOT-OS/RIOT/blob/master/CODE_OF_CONDUCT.md)
+Before opening a new feature request, check the
+[existing feature requests][existing-feature-request] if there's one already
+opened on the same topic.
+
+To request new features or enhancements, just open a new
+[feature request issue][new-feature-request]. Describe your needs and use
+cases. Give detail on why this is important and/or necessary for RIOT.
+
+[existing-feature-request]: https://github.com/RIOT-OS/RIOT/issues?q=state:open+type:issue+label:"Feature+Request"
+[new-feature-request]: https://github.com/RIOT-OS/RIOT/issues/new?template=feature_request.md
+
+## Bug Reports
+[bug-reports]: #bug-reports
+
+While bugs are unfortunate, they're a reality in software. We can't fix what we
+don't know about, so please report liberally. If you're not sure if something
+is a bug or not, feel free to file a bug anyway.
+
+**If you believe reporting your bug publicly represents a security risk to
+RIOT users, please send an email describing the bug to security@riot-os.org**.
+We would appreciate waiting until the next but one release to fix the bug
```

Comment:
"next but one"?

Reply:
Yeah that sounds bizarre but I trust the native speaker there :)

Reply:
If that's unclear, then I think we should change it. How about "the release after next"? Or, "waiting two releases"?
The idea of two releases is that if we're in the middle of a feature freeze, for example, we won't be able to fix it in time for the next one.

Reply:
Regarding the wording I would prefer "waiting two releases". Though my English is pretty well, I'm already struggling doing head acrobatics what exactly that means.

Regarding procedure I disagree: bugs should always be backported, especially during feature freeze. That's what feature freeze is for, fixing bugs that occur are found while using the fresh release candidate. Ideally (but from experience I think we don't have the manpower of that), hotfixes (i.e. bugfixes that are atomic) found in master would also be backported to the release branch *after* the release and would -- depending on severity -- also cause new sub-releases (e.g. 2018.10.1).

Reply:
wouldn't it be better to propose a standard time interval, i.e. weeks or months, instead of tying it to the release cycle - its also easier to understand? Further, this also points to the fact that we do not have any backporting strategy what so ever in place, to ensure that security bugs/flaws are fixed in previous releases.

Reply:
Do we currently have any process in place for fixing security bugs? Perhaps now is the time to talk some more about this. I propose we bring this up in the meeting tomorrow afternoon.

Reply:
+1 for discussing this in our internal meeting tomorrow, but also we should move this discussion to the maintainer mailing list (or even devel).